### PR TITLE
[SPARK-47714][SQL] Set spark.sql.legacy.timeParserPolicy to CORRECTED by default

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

This config was added in 3.0 and it's time to use the CORRECTED behavior by default. This is not a big deal as the CORRECTED behavior is also throwing an error, but the error message is improved to show the actual problem.

### Why are the changes needed?

better error message

### Does this PR introduce _any_ user-facing change?

error message becomes different

### How was this patch tested?

existing tests

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

**Comment by pan3793** [comment]
migration guide?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -4022,7 +4022,7 @@ object SQLConf {
     .stringConf
     .transform(_.toUpperCase(Locale.ROOT))
     .checkValues(LegacyBehaviorPolicy.values.map(_.toString))
-    .createWithDefault(LegacyBehaviorPolicy.EXCEPTION.toString)
+    .createWithDefault(LegacyBehaviorPolicy.CORRECTED.toString)
```
Comment (yaooqinn) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:4025]: Please also refresh the config doc field

**Review by yaooqinn — APPROVED** [review]
LGTM, only one minor comment https://github.com/apache/spark/pull/45853#discussion_r1549909335

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM with the same above doc comments (migration guide and config doc field).

Also, please fix all test case failures like the following.
```
[info] TimestampFormatterSuite:
[info] - explicitly forbidden datetime patterns (13 milliseconds)
[info] - SPARK-31939: Fix Parsing day of year when year field pattern is missing *** FAILED *** (4 milliseconds)
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala (hunk 1 above)
Comment (gengliangwang) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:4025]: Yes let's update the doc "The default value is EXCEPTION..."

**Comment by cloud-fan** [comment]
closing in favor of https://github.com/apache/spark/pull/45859

**Pull request closed without merging**